cdap-client/src/main/java/co/cask/cdap/client/config/ClientConfig.java

```diff
@@ -103,16 +105,15 @@ public URL resolveURL(String apiVersion, String namespace, String path) throws M
   }
 
   /**
-   * Resolves a path against the target CDAP server with the provided apiVersion
+   * Resolves a path against the target CDAP server with the provided namespace, using V3 APIs
    *
-   * @param apiVersion the api version to use
    * @param path Path to the HTTP endpoint. For example, "apps" would result
-   *             in a URL like "http://example.com:10000/v2/apps".
+   *             in a URL like "http://example.com:10000/v3/<namespace>/apps".
    * @return URL of the resolved path
    * @throws MalformedURLException
    */
-  public URL resolveURL(String apiVersion, String path) throws MalformedURLException {
-    return getBaseURI().resolve("/" + apiVersion + "/" + path).toURL();
+  public URL resolveURLV3(String path) throws MalformedURLException {
+    return resolveURL(Constants.Gateway.API_VERSION_3_TOKEN, namespace, path);
```

Comment:
Think we should have the clients explicitly specify the version token, but probably in another PR.